internal/pkg/agent/install/install_windows.go

```diff
@@ -55,12 +55,17 @@ func fixInstallMarkerPermissions(markerFilePath string, ownership utils.FileOwne
 }
 
 // withServiceOptions just sets the user/group for the service.
-func withServiceOptions(username string, groupName string) ([]serviceOpt, error) {
+func withServiceOptions(username string, groupName string, password string) ([]serviceOpt, error) {
 	if username == "" {
 		// not installed with --unprivileged; nothing to do
 		return []serviceOpt{}, nil
 	}
 
+	if password != "" {
+		// existing user
+		return []serviceOpt{withUserGroup(username, groupName), withPassword(password)}, nil
```

Comment:
At this point the `username` must either have `.\` at the start or the DOMAIN NAME like `AD\username-in-ad`. You are going to need to handle username having `AD\` or not having `.\` in the name. Otherwise the service will not start.